packages/utils/src/debounce.js

```diff
@@ -0,0 +1,10 @@
+let timer;
+
+export const debounce = (func, duration) => {
+  return (...args) => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      func(...args);
+    }, duration);
+  };
+};
```

Comment:
Looks like we've got a whole bunch of ways we do debouncing throughout the codebase 😅 
I'd probably argue we should just standardise to use `lodash.debounce` (we already use it in the admin-panel) rather than custom code? Sorry to be _that guy_ 🙏

Reply:
I've tried `lodash.debounce` and it didn't work as expected. Sadly the timer in `lodash.debounce` is somehow being recreated, even with `useCallback()` .. 

Here I've pull out the timer to make sure it is persisted.

Reply:
Ahh that's annoying 😢 

Now that you mention it though, I've got a bit of a concern around this `debounce`. What happens if there are multiple separate uses of it? Won't they share the same `timer`, which will break one of them?

Reply:
Good thinking and thanks! I just found that's an existing bug.. Have found a way to persist timer so we can use `lodash.debounce` here, see the latest commit.